pgjdbc/src/main/java/com/yugabyte/ysql/YBManagedHostnameVerifier.java

```diff
@@ -0,0 +1,221 @@
+package com.yugabyte.ysql;
+
+import org.postgresql.core.PGStream;
+import org.postgresql.jdbc.PgConnection;
+import org.postgresql.util.GT;
+import org.postgresql.util.HostSpec;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.postgresql.util.internal.Nullness.castNonNull;
+
+public class YBManagedHostnameVerifier implements HostnameVerifier {
+
+  private static final Logger LOGGER = Logger.getLogger(YBManagedHostnameVerifier.class.getName());
+  protected static final String GET_SERVERS_QUERY = "select * from yb_servers()";
+  protected Boolean useHostColumn = null;
+
+
+
+
+  private static final int TYPE_DNS_NAME = 2;
+  private static final int TYPE_IP_ADDRESS = 7;
+  private final Properties originalProperties;
+  protected ArrayList<String> currentPublicIps = new ArrayList<>();
+  protected Map<String, String> hostPortMap = new HashMap<>();
+  protected Map<String, String> hostPortMapPublic = new HashMap<>();
+  private final PGStream stream;
+  private Connection controlConnection = null;
```

Comment:
Is a new verifier instance created for each verify() call? If yes, we'll need the control connection to be a static member.